openvdb/openvdb/version.h.in

```diff
@@ -6,83 +6,85 @@
 #define OPENVDB_VERSION_HAS_BEEN_INCLUDED
 
-#include "Platform.h"
+#include "openvdb/Platform.h"
+
 #include <cstddef> // size_t
 #include <cstdint> // uint32_t
 
+///////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
 
-// Library major, minor and patch version numbers
-#define OPENVDB_LIBRARY_MAJOR_VERSION_NUMBER 8
-#define OPENVDB_LIBRARY_MINOR_VERSION_NUMBER 1
-#define OPENVDB_LIBRARY_PATCH_VERSION_NUMBER 0
-
-// If OPENVDB_ABI_VERSION_NUMBER is already defined (e.g., via -DOPENVDB_ABI_VERSION_NUMBER=N)
-// use that ABI version.  Otherwise, use this library version's default ABI.
-#ifdef OPENVDB_ABI_VERSION_NUMBER
-    #if OPENVDB_ABI_VERSION_NUMBER > OPENVDB_LIBRARY_MAJOR_VERSION_NUMBER
-        // If using a future OPENVDB_ABI_VERSION_NUMBER, issue a message directive.
-        // This can be optionally suppressed by setting the CMake option
-        // OPENVDB_USE_FUTURE_ABI_<VERSION>=ON.
-        #if OPENVDB_ABI_VERSION_NUMBER == 9
-            #ifndef OPENVDB_USE_FUTURE_ABI_9
-                PRAGMA(message("NOTE: ABI = 9 is still in active development and has not been finalized, "
-                    "CMake option OPENVDB_USE_FUTURE_ABI_9 suppresses this message"))
-            #endif
-        #else
-            #error expected OPENVDB_ABI_VERSION_NUMBER <= OPENVDB_LIBRARY_MAJOR VERSION_NUMBER
-        #endif
-    #endif
-#else
-    #define OPENVDB_ABI_VERSION_NUMBER OPENVDB_LIBRARY_MAJOR_VERSION_NUMBER
-#endif
+/// @brief  Auto generated build configurations from CMake
 
-// If using an OPENVDB_ABI_VERSION_NUMBER that has been deprecated, issue a message
-// directive. Note that an error is also set in openvdb.cc which enforces stricter
-// behavior during compilation of the library. Both can be optionally suppressed
-// by defining OPENVDB_USE_DEPRECATED_ABI_<VERSION>.
-#ifndef OPENVDB_USE_DEPRECATED_ABI_6
-    #if OPENVDB_ABI_VERSION_NUMBER == 6
-        PRAGMA(message("NOTE: ABI = 6 is deprecated, CMake option OPENVDB_USE_DEPRECATED_ABI_6 "
-            "suppresses this message"))
-    #endif
+/* Library major, minor and patch version numbers */
+#define OPENVDB_LIBRARY_MAJOR_VERSION_NUMBER ${OpenVDB_MAJOR_VERSION}
+#define OPENVDB_LIBRARY_MINOR_VERSION_NUMBER ${OpenVDB_MINOR_VERSION}
+#define OPENVDB_LIBRARY_PATCH_VERSION_NUMBER ${OpenVDB_PATCH_VERSION}
+
+/* The ABI version that OpenVDB was built with */
+/// @note  This ifndef exists for compatibility with older versions of OpenVDB.
+///   This value should never be different from the value configured when
+///   OpenVDB was built, but this previously needed to be defined by downstream
+///   software. Redefining it here would cause build failures, so this allows
+///   users to transition and remove the define in their build systems.
+#ifndef OPENVDB_ABI_VERSION_NUMBER
+#define OPENVDB_ABI_VERSION_NUMBER ${OPENVDB_ABI_VERSION_NUMBER}
 #endif
```

Comment:
Added `OPENVDB_ABI_VERSION_NUMBER` as an auto generated value to `version.h`. Note that, this is only defined if it's not previously defined (as most people will be defining this in their software and we'd get duplicate definition errors/warnings). Going forward, users won't need to worry about this value.

I did consider doing this:

```
#ifndef OPENVDB_ABI_VERSION_NUMBER
#define OPENVDB_ABI_VERSION_NUMBER ${OPENVDB_ABI_VERSION_NUMBER}
#else
    #if OPENVDB_ABI_VERSION_NUMBER != ${OPENVDB_ABI_VERSION_NUMBER}
        #warning "Specified OpenVDB ABI value does not match built configuration!" // could also error
    #endif
#endif
```

But decided against it for now